src/Tribe/PUE/Checker.php

```diff
@@ -244,51 +246,60 @@ public function is_key_valid() {
 		 * Iterate on all the registered PUE Product Licenses we have and find if any are valid.
 		 * Will revalidate the licenses if none are found to be valid.
 		 *
-		 * @todo In scenarios where a user goes from a Free license to an active license the transient may give a false positive.
-		 *
 		 * @since 6.3.2
+		 * @since TBD Refactored logic to take into account the transient.
 		 *
 		 * @return bool
 		 */
 		public static function is_any_license_valid(): bool {
 			$valid_slug = 'valid';
-			$has_valid  = false;
+			$invalid_slug = 'invalid';
 
-			// Check our transient.
+			// Check the transient value first.
 			$transient_value = get_transient( self::IS_ANY_LICENSE_VALID_TRANSIENT_KEY );
-			if ( ! empty( $transient_value ) ) {
-				return $transient_value === $valid_slug;
+
+			if ( $transient_value === $valid_slug ) {
+				// Transient explicitly marked as valid.
+				return true;
+			}
+
+			if ( $transient_value === $invalid_slug ) {
+				// Transient explicitly marked as invalid, no recheck.
+				return false;
 			}
 
-			// Check our local transient/cache first.
+			if ( self::has_valid_uplink_license() ) {
+				set_transient( self::IS_ANY_LICENSE_VALID_TRANSIENT_KEY, $valid_slug, HOUR_IN_SECONDS );
+				return true;
```

Comment:
Because we are checking now if did_action that could only be true when=>

- Current request updated an uplink license
- is_any_license_valid is being executed after license validation - which i dont think is the case

So i would prefer to rather add a listener for the uplink license updated rather than checking if it was done prior to this.